topologytest/hlv_test.go

```diff
@@ -187,18 +400,30 @@ func stripInternalProperties(body db.Body) {
 	delete(body, "_id")
 }
 
-func waitForVersionAndBody(t *testing.T, testCase singleActorTest, peers map[string]Peer, expectedVersion DocMetadata, expectedBody []byte) {
+func waitForVersionAndBody(t *testing.T, testCase ActorTest, peers map[string]Peer, expectedVersion DocMetadata, expectedBody []byte) {
 	// sort peer names to make tests more deterministic
 	peerNames := maps.Keys(peers)
 	for _, peerName := range peerNames {
 		peer := peers[peerName]
-		t.Logf("waiting for doc version on %s, written from %s", peer, testCase.activePeerID)
-		body := peer.WaitForDocVersion(testCase.collectionName(), testCase.docID(), expectedVersion)
+		t.Logf("waiting for doc version on %s, written from %s", peer, testCase.getOriginalActor())
+		body := peer.WaitForDocVersion(testCase.collectionName(), testCase.docID(testCase.getOriginalActor()), expectedVersion)
+		requireBodyEqual(t, expectedBody, body)
+	}
+}
+
+// waitForVersionAndBodyOnNonActivePeers waits for a document version on all peers that the update didn't occur on
+func waitForVersionAndBodyOnNonActivePeers(t *testing.T, testCase ActorTest, peers map[string]Peer, expectedVersion DocMetadata, expectedBody []byte, writingPeer string) {
+	delete(peers, writingPeer)
```

Comment:
A map is a pointer here, so I might have the caller filter out the peer names since this worries me to do a modification. You should be able to just do `slices.Delete(maps.Keys(peerNames), peerNames)`.